ls1intum/Artemis
# Feature/grading instructions/modify markdown editor for sgi

### Checklist

### Motivation and Context


Given the current situation, instructors add a block of text describing the grading instructions in the text editor. However, in order to unify the assessment feedback, we have to divide the text block in specific parameters, that could, later on, be dragged and dropped by tutors when assessing a submission.
### Description

Divided an instruction into the following parameters :
- Credits
- Grading Scale
- Grading Instruction Description
- Feedback
- Usage Count (which indicates how often the points should be applied for the instruction)

The GradingInstructionCommand is added to enable users to add all parameters of an instruction.
Instruction could be stand-alone or part of a criterion. The opening identifier `[gradingCriterion]` indicates that all instructions below belong to this criterion.
### Steps for Testing

1. Navigate to Course Administration
2. Navigate to Exercise
3. Edit Modeling/text/file-upload exercises
4. Add free text in the grading instructions editor and save it
5. Switch to structured Grading instructions -> if you switch back to grading instructions the free text inserted before should still be available
6. Add a new criterion/ edit existing criterion/ delete an existing criterion
7. Add one or multiple instructions to a criterion of your choice/ edit and delete it
8. Add/edit and delete stand-alone instructions
9. Save your changes -> when you click on edit again your changes should be displayed correctly

### Screenshots


stand-alone instruction:
![image](https://user-images.githubusercontent.com/50635005/76216735-695ead80-6211-11ea-9d89-c43e310944de.png)

stand-alone instruction and instructions that belong to a criterion
![image](https://user-images.githubusercontent.com/50635005/76216850-9c08a600-6211-11ea-8416-8119ee8ed4d3.png)

SGI Preview
![image](https://user-images.githubusercontent.com/50635005/76247761-5adbba80-6240-11ea-90cf-16fd18005b80.png)
